pylint/checkers/match_statements_checker.py

```diff
@@ -124,13 +176,27 @@ def check_duplicate_sub_patterns(
             attrs.add(name)
 
     @only_required_for_messages(
+        "match-class-positional-attributes",
         "multiple-class-sub-patterns",
         "too-many-positional-sub-patterns",
     )
     def visit_matchclass(self, node: nodes.MatchClass) -> None:
         attrs: set[str] = set()
         dups: set[str] = set()
 
+        if node.patterns:
+            if isinstance(node, nodes.MatchClass) and isinstance(node.cls, nodes.Name):
+                inferred = safe_infer(node.cls)
+                if not (
+                    isinstance(inferred, nodes.ClassDef)
+                    and inferred.qname() in MATCH_CLASS_SELF_NAMES
+                ):
+                    self.add_message(
+                        "match-class-positional-attributes",
+                        node=node,
+                        confidence=HIGH,
+                    )
```

Comment:
Looking at the `sentry` results, I do wonder if this is to strict and we'd end up emitting too many false positives. `self` matches also work for every subclass of the special builtin classes. Most notably `NamedTuple`.